.github/workflows/deploy_to_environment_azure.yml

```diff
@@ -0,0 +1,171 @@
+name: Deploy to environment (CIP)
+
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: "Environment (staging, production)"
+        required: true
+
+env:
+  DOCKER_IMAGE: a2bint-app
+  NODE_VERSION: 18.x
+
+jobs:
+#   test:
+#     runs-on: ubuntu-22.04
+#     container:
+#       image: node:14.15.0-alpine3.12
+#     steps:
+#       - name: Test the build
+#         run: |
+#           dotnet test -c Release ApplyToBecomeInternal/ApplyToBecomeInternal.sln --no-build --verbosity normal  --collect:"XPlat Code Coverage"
+
+  build-and-push-image:
+    needs: test
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set environment variables
+        run: |
+          ENVIRONMENT=${{ github.event.inputs.environment }^^}
+
+          if [ "$ENVIRONMENT" == "STAGING" ]; then
+            ACR_CLIENTID=${{ secrets.TEST_AZURE_ACR_CLIENTID }}
+            ACR_SECRET=${{ secrets.TEST_AZURE_ACR_SECRET }}
+            ACR_URL=${{ secrets.TEST_ACR_URL }}
+          elif [ "$ENVIRONMENT" == "PRODUCTION"]; then
+            ACR_CLIENTID=${{ secrets.PRODUCTION_AZURE_ACR_CLIENTID }}
+            ACR_SECRET=${{ secrets.PRODUCTION_AZURE_ACR_SECRET }}
+            ACR_URL=${{ secrets.PRODUCTION_ACR_URL }}
+          else
+              echo "$ENVIRONMENT doesn't match any known options"
+              exit 1
+          fi
+
+          echo "ACR_CLIENTID=$ACR_CLIENTID" >> $GITHUB_ENV
+          echo "ACR_SECRET=$ACR_SECRET" >> $GITHUB_ENV
+          echo "ACR_URL=$ACR_URL" >> $GITHUB_ENV
+
+          GIT_REF=${{ github.ref }}
+          GIT_BRANCH=${GIT_REF##*/}
+          echo "BRANCH_TAG=$GIT_BRANCH" >> $GITHUB_ENV
+          echo "DOCKER_IMAGE_TAG=$GITHUB_SHA" >> $GITHUB_ENV
+
+      - name: Azure Container Registry login
+        uses: docker/login-action@v2
+        with:
+          username: ${{ env.ACR_CLIENTID }}
+          password: ${{ env.ACR_SECRET }}
+          registry: ${{ env.ACR_URL }}
+
+      - name: Build and push docker image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: Dockerfile.gpaas-azure-migration
+          tags: |
+            ${{ env.ACR_URL }}/${{ env.DOCKER_IMAGE }}:${{ env.DOCKER_IMAGE_TAG }}
+            ${{ env.ACR_URL }}/${{ env.DOCKER_IMAGE }}:${{ env.BRANCH_TAG }}
+          push: true
+          build-args: |
+            COMMIT_SHA=${{ env.DOCKER_IMAGE_TAG }}
+
+  deploy-image:
+    needs: build-and-push-image
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Set environment variables
+        run: |
+          ENVIRONMENT=${{ github.event.inputs.environment }^^}
+
+          echo "Deploying to $ENVIRONMENT environment"
+
+          if [ "$ENVIRONMENT" == "STAGING" ]; then
```

Comment:
You might be uppercasing the environment name somewhere and I can't see it but wanted to make sure you were.

Reply:
[#L34](https://github.com/DFE-Digital/a2b-internal/blob/ops/github-action-workflow/.github/workflows/deploy_to_environment_azure.yml#L34) uses `^^` which is a `bash` way of Uppercasing a string. (untested in the context of a github action though)